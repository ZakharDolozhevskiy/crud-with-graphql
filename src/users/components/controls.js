import React from 'react';
import styled from 'styled-components';

import Switch from '@material-ui/core/Switch';
import CreateUser from './create-user';
import FormControlLabel from '@material-ui/core/FormControlLabel';

export class Controls extends React.PureComponent {
  render() {
    return (
      <header className={this.props.className}>
        <CreateUser/>
        <FormControlLabel
          label="Show only active users"
          classes={{ label: 'active-user-label' }}
          control={
            <Switch
              color="primary"
              onChange={this.props.onChange}
              checked={this.props.showActive}
            />}
        />
      </header>
    )
  }
}

export default styled(Controls)`
  .active-user-label {
    font-size: 16px !important;
  }
`;