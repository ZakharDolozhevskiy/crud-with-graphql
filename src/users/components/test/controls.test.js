import React from 'react';
import toJSON from 'enzyme-to-json';
import { shallow } from 'enzyme';

import { Controls } from '../controls';

describe('<Controls/>', () => {
  const onChange = jest.fn();
  const component = shallow(<Controls showActive onChange={onChange} />);

  it('renders controls component in active state', () => {
   expect(toJSON(component)).toMatchSnapshot();
  });

  it('renders controls component in inactive state', () => {
    component.setProps({ showActive: false });
    expect(toJSON(component)).toMatchSnapshot();
  });
});