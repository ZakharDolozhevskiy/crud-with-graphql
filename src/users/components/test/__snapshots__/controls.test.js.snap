// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Controls/> renders controls component in active state 1`] = `
<header>
  <Styled(CreateUser) />
  <WithStyles(FormControlLabel)
    classes={
      Object {
        "label": "active-user-label",
      }
    }
    control={
      <WithStyles(Switch)
        checked={true}
        color="primary"
        onChange={[Function]}
      />
    }
    label="Show only active users"
  />
</header>
`;

exports[`<Controls/> renders controls component in inactive state 1`] = `
<header>
  <Styled(CreateUser) />
  <WithStyles(FormControlLabel)
    classes={
      Object {
        "label": "active-user-label",
      }
    }
    control={
      <WithStyles(Switch)
        checked={false}
        color="primary"
        onChange={[Function]}
      />
    }
    label="Show only active users"
  />
</header>
`;
